### PR TITLE
fix: Added Non-Finite check

### PR DESCRIPTION
**Pull request by dubeyaditya29**
This PR fixes the missing check for non-finite values being added to the map object in JSONObject 

Added a new method `isFiniteNumber(Object value)`  returns a boolean value false in case the value is non-finite.
The wrap throws a runtime exception in case the value is non-finite.

### CODE REVIEW

**Comment by dubeyaditya29** [comment]
#713

**Comment by stleary** [comment]
@dubeyaditya29 Good start, but more work is needed.

Please revert the change in pom.xml:
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project json: Fatal error compiling: invalid target release: 9 -> [Help 1]

Are there any other numeric types that should be checked?
Does JSONArray need this check too?
Each new line of code should receive coverage from one or more of the new unit tests.

File: src/main/java/org/json/JSONObject.java [hunk 1]
```diff
@@ -2629,8 +2632,34 @@ public static Object wrap(Object object) {
         return wrap(object, null);
     }
 
+    // Custom isFiniteNumber method
+    public static boolean isFiniteNumber(Object value) {
```
Comment (johnjaylward) [inline src/main/java/org/json/JSONObject.java:2636]: This method is sub-optimal, and redundant of `testValidity` and `numberIsFinite` methods that are already available. Please use the existing function(s)

File: pom.xml [hunk 2]
```diff
@@ -95,8 +95,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.11.0</version>
                 <configuration>
-                    <source>1.8</source>
```
Comment (johnjaylward) [inline pom.xml:98]: As stated already, this needs to be reverted. Java 1.8 is our current target.

File: src/main/java/org/json/JSONObject.java [hunk 3]
```diff
@@ -2629,8 +2632,34 @@ public static Object wrap(Object object) {
...
+            return Double.isFinite(doubleValue);
+        } else {
+            // Value is not a Number
+            return false;
+        }
+    }
     private static Object wrap(Object object, Set<Object> objectsRecord) {
         try {
+            // Calling the isFiniteNumber(Object object) -> to check if the object is Non-finite.
+            if(!isFiniteNumber(object)){
```
Comment (johnjaylward) [inline src/main/java/org/json/JSONObject.java:2660]: this check should happen after the `NULL` check.

**Review by johnjaylward — CHANGES_REQUESTED** [review]

File: src/main/java/org/json/JSONObject.java [hunk 4]
```diff
@@ -2629,8 +2632,34 @@ public static Object wrap(Object object) {
         return wrap(object, null);
     }
 
+    // Custom isFiniteNumber method
+    public static boolean isFiniteNumber(Object value) {
+        if (value instanceof Number) {
+            double doubleValue;
+            if (value instanceof Integer) {
+                doubleValue = ((Integer) value).doubleValue();
```
Comment (Madjosz) [inline src/main/java/org/json/JSONObject.java:2640]: Integers are always finite.

Comment (johnjaylward) [reply src/main/java/org/json/JSONObject.java:2640]: agreed. Using the existing `testValidity` method would be the correct course and already covers what this PR is attempting.

**Comment by dubeyaditya29** [comment]
hey @stleary Thankyou for reviewing this.
1. Have added Integer, Float, Double, I will add Long as well, let me know if missing any specific type.
2. Let me add test cases for that as well will get back on this.
3. For the last point I have created a different test class file for checking the JSONObject, let me know if I have missed something will add that too.

File: pom.xml (hunk 2 above)
Comment (dubeyaditya29) [reply pom.xml:98]: I'll made the necessary changes here thanks.

File: src/main/java/org/json/JSONObject.java (hunk 4 above)
Comment (dubeyaditya29) [reply src/main/java/org/json/JSONObject.java:2640]: agree @johnjaylward  the testValidity can alone verify this, am somehow new to this code base, can you help me understand how did you found this method.

**Comment by dubeyaditya29** [comment]
thanks to @johnjaylward suggestions I think there is no need for defining a new method here we already have `testValidaity` which can fulfil the requirements.

File: src/test/java/org/json/junit/data/JsonObjectTest.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package org.json.junit.data;
```
Comment (Madjosz) [inline src/test/java/org/json/junit/data/JsonObjectTest.java:1]: Tests should be added to the existing class `org.json.junit.JSONObjectTest`. I would suggest to name the methods `issue713...()` analogously to already existing issue-related test methods in that class.

File: src/main/java/org/json/JSONObject.java [hunk 6]
```diff
@@ -2629,8 +2632,34 @@ public static Object wrap(Object object) {
...
+        } else {
+            // Value is not a Number
+            return false;
+        }
+    }
     private static Object wrap(Object object, Set<Object> objectsRecord) {
         try {
+            // Calling the isFiniteNumber(Object object) -> to check if the object is Non-finite.
+            if(!isFiniteNumber(object)){
+                throw new RuntimeException("Non-Finite Number encountered");
```
Comment (Madjosz) [inline src/main/java/org/json/JSONObject.java:2661]: Throw `JSONException`, not a generic `RuntimeException`.

**Comment by johnjaylward** [comment]
#779 is a better implementation. This should be rejected

**Comment by stleary** [comment]
Thanks, but not accepting this PR.

**Comment by dubeyaditya29** [comment]
No problem @stleary thanks for reviewing.

**Comment by stleary** [comment]
@dubeyaditya29 if you want to work on a different issue, I recommend #748, which I just reopened. Let me know if you have any questions.

**Pull request closed without merging**